ecobee/auth.go

```diff
@@ -60,7 +60,7 @@ func (ts *tokenSource) save() error {
 	return err
 }
 
-func (ts *tokenSource) firstAuth() error {
+func (ts *tokenSource) FirstAuth() (PinResponse, error) {
```

Comment:
I am changing this as requested. Counter-argument anyways: callers should always check an error before using the normal return value. No need to use a pointer for a simple struct. Still new to Go, does that seem like a valid argument?

Reply:
fixed

Reply:
I'm not finding the reference I was looking for.  But I believe the rationale is to avoid the unnecessary memory allocation.  (An unnecessary micro-optimization in this case.)  More importantly, this is the common pattern.  Structs are almost always passed by reference.